datastore/modelmanager.go

```diff
@@ -462,3 +474,21 @@ func (db *DB) checkBoolQuery(row *sql.Row) bool {
 
 	return found
 }
+
+// defaultAPIKey creates a default API key if the field is empty
+func (db *DB) defaultAPIKey(model *Model) error {
+	// Remove all whitespace from the API key
+	model.APIKey = strings.Replace(model.APIKey, " ", "", -1)
+
+	// Check we have a minimum API key size
+	if len(model.APIKey) > minAPIKeyLength {
+		return nil
+	}
+
+	apiKey, err := db.generateAPIKey()
```

Comment:
why generateAPIKey() is a db method and not a standalone one?
If it can be a standalone one, also defaultAPIKey() could be